### PR TITLE
Basic Server Behavior

### PR DESCRIPTION
**Pull request by tmccurbin**

### CODE REVIEW

File: media-player.js [hunk 1]
```diff
@@ -1,28 +1,131 @@
...
+    }
+    
+    // Determine whether files are audio, video, or unsupported
+    getMediaTypes() {
+        this.playlist.forEach((url) => {
+            let extension = path.parse(url).ext.toLowerCase();
+            if (audioTypes.has(extension)) this.mediaTypes.push('audio');
+            else if (videoTypes.has(extension)) this.mediaTypes.push('video');
+            else if (ambiguousTypes.has(extension)) {
+                const shellCommand = `ffmpeg -i ${url} -hide_banner 2>&1 | grep `;
```
Comment (itsrachelfish) [inline media-player.js]: Interesting use of ffmpeg to determine media type

File: media-player.js [hunk 2]
```diff
@@ -1,28 +1,131 @@
...
+    startTimers() {
+        this.startTime = new Date(); // Start main timer
+        for (let index = 0; index < this.breakpoints.length; index++) {
+            let breakpointMillisecs = this.breakpoints[index] * 1000;
+            
+            // Set timers to update mediaIndex and notify users of next URL in playlist
+            if (index === (this.breakpoints.length - 1)) {
+                setTimeout(() => {
+                    // Emit socket event here
+                    let url = this.playlist[0].slice(8);
```
Comment (itsrachelfish) [inline media-player.js]: Why are you using `slice(8)` here?

File: server.js [hunk 3]
```diff
@@ -1,29 +1,50 @@
...
 // Socket setup
-let io = socket(server)
-io.on('connection', () => {
-    console.log(`Made socket connection with a client`)
-})
\ No newline at end of file
+let io = socket(server);
+io.on('connection', (client) => {
+    let index = mediaPlayer.mediaIndex;
+    let url = mediaPlayer.playlist[index].slice(8);
```
Comment (itsrachelfish) [inline server.js]: Same question as above, why the `slice(8)`?

File: media-player.js (hunk 2 above)
Comment (tmccurbin) [reply media-player.js]: While the server methods (ffmpeg and ffrpobe) use the path relative to root directory, the client uses the path relative to '/public'.

File: server.js (hunk 3 above)
Comment (tmccurbin) [reply server.js]: Same as above: the client uses the path relative to '/public'

File: media-player.js (hunk 2 above)
Comment (itsrachelfish) [reply media-player.js]: Hmm, I feel like it would be more readable to add the string '/public' when necessary instead of slicing it off.

**Pull request merged**